pkg/updater/updater.go

```diff
@@ -213,29 +213,30 @@ func (u *updaterImpl) BootstrapURL(ctx context.Context, url string) error {
 	defer u.m.Unlock()
 	u.refreshState(ctx)
 	defer u.refreshState(ctx)
-	stablePackage, err := u.downloader.Package(ctx, url)
-	if err != nil {
-		return fmt.Errorf("could not get package from url: %w", err)
-	}
-	return u.boostrapPackage(ctx, stablePackage)
+
+	return u.boostrapPackage(ctx, url, "", "")
 }
 
-func (u *updaterImpl) boostrapPackage(ctx context.Context, stablePackage Package) error {
+func (u *updaterImpl) boostrapPackage(ctx context.Context, url string, expectedPackage string, expectedVersion string) error {
 	// both tmp and repository paths are checked for available disk space in case they are on different partitions
 	err := checkAvailableDiskSpace(fsDisk, defaultRepositoriesPath, os.TempDir())
 	if err != nil {
 		return fmt.Errorf("not enough disk space to install package: %w", err)
 	}
-	log.Infof("Updater: Bootstrapping stable version %s of package %s", stablePackage.Version, stablePackage.Name)
-	image, err := u.downloader.Download(ctx, stablePackage)
+	log.Infof("Updater: Bootstrapping stable package from %s", url)
+	downloadedPackage, err := u.downloader.Download(ctx, url)
 	if err != nil {
 		return fmt.Errorf("could not download: %w", err)
 	}
-	err = u.installer.installStable(stablePackage.Name, stablePackage.Version, image)
+	if (expectedPackage != "" && downloadedPackage.Name != expectedPackage) || (expectedVersion != "" && downloadedPackage.Version != expectedVersion) {
```

Comment:
why not enforce the version on the download (err in it, and run a retry actually since this happens on release)?

Reply:
No, this won't happen anymore with this PR. The issue with the previous implementation was that we were downloading the manifest 2 times using the `latest` label and that the underlying manifest could change between the two. Now we only grab the manifest once.